src/Microsoft.AspNetCore.OData/Adapters/WebApiUrlHelper.cs

```diff
@@ -17,42 +24,49 @@ namespace Microsoft.AspNet.OData.Adapters
     internal class WebApiUrlHelper : IWebApiUrlHelper
     {
         /// <summary>
-        /// The inner helper wrapped by this instance.
+        /// The inner request wrapped by this instance.
         /// </summary>
-        internal IUrlHelper innerHelper;
+        internal HttpRequest innerRequest;
 
         /// <summary>
         /// Initializes a new instance of the WebApiUrlHelper class.
         /// </summary>
         /// <param name="helper">The inner helper.</param>
-        public WebApiUrlHelper(IUrlHelper helper)
+        public WebApiUrlHelper(HttpRequest request)
```

Comment:
The class named "UrlHelper", why do you want to wrap a "HttpRequest"?

Reply:
I had it that way at some point and refactored it decouple Request and Url helper since they are not as closely related in AspNetCore as they are in AspNet. Request is used in WebApiUrlHelper more than UrlHelper is and it was handle at the time of construction.

However, it's a simple refactor to make this IUrlHelper, thanks for the comment.